niworkflows/viz/utils.py

```diff
@@ -66,6 +69,65 @@ def save_html(template, report_file_name, unique_string, **kwargs):
         handle.write(report_render)
 
 
+def svg_compress(image, compress='auto'):
+    ''' takes an image as created by nilearn.plotting and returns a blob svg.
+    Performs compression (can be disabled). A bit hacky. '''
+
+    # Compress the SVG file using SVGO
+    if (shutil.which("svgo") and compress == 'auto') or compress is True:
+
+        p = subprocess.Popen("svgo -i - -o - -q -p 3 --pretty --disable=cleanupNumericValues",
+                             stdin=subprocess.PIPE,
+                             stdout=subprocess.PIPE,
+                             shell=True)
+        image = p.communicate(input=image.encode('utf-8'))[0].decode('utf-8')
+        if p.wait() != 0:
+            raise Exception("svgo failed to run.")
```

Comment:
Could drop the manual exception by using `check=True`, and you can use `input` instead of `stdin` to pass a `bytes` directly.

```Python
p = subprocess.run("svgo -i - -o - -q -p 3 --pretty --disable=cleanupNumericValues",
                   input=image.encode('utf-8'), stdout=subprocess.PIPE,
                   shell=True, check=True)
image = p.stdout.decode('utf-8')
```

Failure should raise a `CalledProcessError`, if that's sufficient.